--- a/trace.h	2025-10-10 22:58:35.000000000 +0200
+++ b/trace.h	2025-10-12 07:14:37.000000000 +0200
@@ -114,7 +114,7 @@
 	),
 
 	TP_fast_assign(
-		strlcpy(__entry->wiphy_name,
+		strscpy(__entry->wiphy_name,
 			wiphy_name(mt76_dev_phy(dev, status->phy_idx)->hw->wiphy),
 			MAXNAME);
 		__entry->seqno = status->seqno;
@@ -143,7 +143,7 @@
 	),
 
 	TP_fast_assign(
-		strlcpy(__entry->wiphy_name,
+		strscpy(__entry->wiphy_name,
 			wiphy_name(dev->phys[wcid->phy_idx]->hw->wiphy),
 			MAXNAME);
 		__entry->wcid = wcid->idx;
